core/src/parking_lot.rs

```diff
@@ -193,6 +208,14 @@ where
     // to construct. Try to use a thread-local version if possible. Otherwise just
     // create a ThreadData on the stack
     let mut thread_data_storage = None;
+
+    // REVIEW: there's a circular dependency between thread locals and
+    // synchronization. On systems that don't have `#[thread_local]` (aka
+    // windows) soemtimes a `Mutex` is used to synchronize creation of keys.
+    // That means that attempting to access the thread local here may acquire a
+    // lock which would in turn come back to access thread data?
+    //
+    // REVIEW: what are the error cases that accessing the thread local fails?
```

Comment:
The error case for `try_with` is when a thread is currently in the middle of destroying its TLS. In such cases we fall back to a stack allocation.

Reply:
Ok makes sense, can this be organized without `unsafe` though?